src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -3232,14 +3232,10 @@ def _remove_osa_nulls(managed_clusters):
     by the server, but get recreated by the CLI's own "to_dict" serialization.
     """
     attrs = ['tags', 'plan', 'type', 'id']
-    ap_master_attrs = ['name', 'os_type']
```

Comment:
It doesn't seem like we ever had `name` and `os_type` attributes in the public API.

Reply:
But this code been here since the beginning of times too. Not sure what broke the command.

Reply:
Hi @m1kola , is this because the different API version? Are you sure we can remove these code safely?

Reply:
@arrownj I'm currently not sure why it's happening. It was working fine on Azure CLI 2.3.1 (maybe on newer versions too) and `az openshift show` and `az openshift list` are broken on 2.5.1 and `dev` branch.

As far as I see, we did not change anything in the API and we never had `name` and `os_type` fields in the master pool profile API on the server (I checked all the supported versions in [our codebase](https://github.com/openshift/openshift-azure/tree/c70152ea551524eb7470c19519d01a1d7035ea8d/pkg/api)).

Reply:
Here's the break:

https://github.com/Azure/azure-sdk-for-python/commit/3f6238b2a224df11fa88c6501e078aac96697277#diff-66c3074fc0745e8946eba5783a5f8f32R21

The client library we were using was stale having been generated on an ancient pre-GA API and ^ brings us up to date.